### PR TITLE
add nodes

### PR DESCRIPTION
**Pull request by angad-k**

### CODE REVIEW

File: ProcSDF/Rendering/Shaders/shader_reference.glsl [hunk 1]
```diff
@@ -0,0 +1,283 @@
+// PREAMBLE
```
Comment (ayushgupta138) [inline ProcSDF/Rendering/Shaders/shader_reference.glsl:1]: is this file required?

Comment (angad-k) [reply ProcSDF/Rendering/Shaders/shader_reference.glsl:1]: not really, but yes.

**Comment by lzardy** [comment]
How far along was this PR? Let me know if there is something I can do to help.

**Comment by angad-k** [comment]
Hey @lzardy sorry for the late reply, I am not actively working on the project anymore other than maybe hacking upon something if something strikes me over some weekend. But I'd be more than happy to help you in contributing. Check out the open issues, propose any new changes if you feel like it, and ping me if you want any clarifications.
